tools.py

```diff
@@ -32,41 +37,74 @@ def subset(input_path, output_path, audio_name, align_boundary_words, cer, wer,
-		segment_path = os.path.join(output_path, os.path.basename(audio_path) + '.{channel}-{begin:.06f}-{end:.06f}.wav'.format(**t))
-		audio.write_audio(segment_path, segment, sample_rate, mono = True)
+		segment = signal[slice(t['channel'], 1 + t['channel']) if t['channel'] is not None else ...,
+		int(max(t['begin'] - dilate, 0) * sample_rate): int((t['end'] + dilate) * sample_rate)]
+
+		segment_file_name = os.path.basename(audio_path) + '.{channel}-{begin:.06f}-{end:.06f}.wav'.format(**t)
+		digest = hashlib.md5(segment_file_name.encode('utf-8')).hexdigest()
+		sub_path = [digest[-1:], digest[:2]] if add_sub_paths else []
+		segment_path_components = [output_path] + sub_path + [segment_file_name]
+
+		segment_path = os.path.join(*segment_path_components)
+		os.makedirs(os.path.join(*segment_path_components[:-1]), exist_ok=True)
+		audio.write_audio(segment_path, segment, sample_rate, mono=True)
+
 		if strip_prefix:
-			segment_path = segment_path[len(strip_prefix):] if segment_path.startswith(strip_prefix) else segment_path
-			t['audio_path'] = t['audio_path'][len(strip_prefix):] if t['audio_path'].startswith(strip_prefix) else t['audio_path']
-
-		t = dict(audio_path = segment_path,
-				channel = 0 if len(signal) == 1 else None,
-				begin = 0.0,
-				end = segment.shape[-1] / sample_rate,
-				speaker = t.pop('speaker', None),
-				ref = t.pop('ref', None),
-				hyp = t.pop('hyp', None),
-				cer = t.pop('cer', None),
-				alignment = t.pop('alignment', []),
-				words = t.pop('words', []),
-				meta = t)
+			segment_path = segment_path[len(strip_prefix):] if segment_path.startswith(
+					strip_prefix) else segment_path
+			t['audio_path'] = t['audio_path'][len(strip_prefix):] if t['audio_path'].startswith(strip_prefix) else \
+			t['audio_path']
+
+		t = dict(audio_path=segment_path,
+				audio_name=os.path.basename(segment_path),
+				channel=0 if len(signal) == 1 else None,
+				begin=0.0,
+				end=segment.shape[-1] / sample_rate,
+				speaker=t.pop('speaker', None),
+				ref=t.pop('ref', None),
+				hyp=t.pop('hyp', None),
+				cer=t.pop('cer', None),
+				wer=t.pop('wer', []),
+				alignment=t.pop('alignment', []),
+				words=t.pop('words', []),
+				meta=t)
 
 		prev_audio_path = audio_path
-		transcript_cat.append(t)
+		audio_path_res.append(t)
+	return audio_path_res
+
+def cut(input_path, output_path, sample_rate, mono, dilate, strip, strip_prefix, audio_backend, add_sub_paths):
+	os.makedirs(output_path, exist_ok = True)
 
-	json.dump(transcripts.strip(transcript_cat, strip), open(os.path.join(output_path, os.path.basename(output_path) + '.json'), 'w'), ensure_ascii = False, sort_keys = True, indent = 2)
+	transcript = json.load(open(input_path))
+	print('Segment count: ', len(transcript))
+
+	transcript_by_path = {t['audio_path']: [] for t in transcript}
+	for t in transcript:
+		transcript_by_path[t['audio_path']].append(t)
+
+	print("Unique audio_path count: ", len(transcript_by_path.keys()))
+	with Pool(processes=32) as pool:
```

Comment:
Do not hardcode `num-workers`, please pass it as an argument

Reply:
fixed